### PR TITLE
simp_le: 0.17.0 -> 0.20.0

### PR DESCRIPTION
**Pull request by makefu**
## Description of changes

updates simp_le to 0.20.0 , fixes tests

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/admin/simp_le/default.nix [hunk 1]
```diff
@@ -2,11 +2,11 @@
 
 python3Packages.buildPythonApplication rec {
   pname = "simp_le-client";
-  version = "0.17.0";
+  version = "0.20.0";
 
   src = fetchPypi {
     inherit pname version;
-    sha256 = "0m1jynar4calaffp2zdxr5yy9vnhw2qf2hsfxwzfwf8fqb5h7bjb";
+    sha256 = "sha256-p6+OF8MuAzcdTV4/CvZpjGaOrg7xcNuEddk7yC2sXIE=";
```
Comment (mweinelt) [inline pkgs/tools/admin/simp_le/default.nix:9]:
```suggestion
    hash = "sha256-p6+OF8MuAzcdTV4/CvZpjGaOrg7xcNuEddk7yC2sXIE=";
```

File: pkgs/tools/admin/simp_le/default.nix [hunk 2]
```diff
@@ -2,11 +2,11 @@
 
 python3Packages.buildPythonApplication rec {
   pname = "simp_le-client";
-  version = "0.17.0";
+  version = "0.20.0";
```
Comment (mweinelt) [inline pkgs/tools/admin/simp_le/default.nix:5]:
```suggestion
  version = "0.20.0";
  pyproject = true;
```

To use the PEP517 builder. Then add `setuptools` into `nativeBuildInputs`.

**Review by mweinelt — DISMISSED** [review]
Please call pre and post hooks in the overridden check phase.

File: pkgs/tools/admin/simp_le/default.nix [hunk 3]
```diff
@@ -17,10 +18,13 @@ python3Packages.buildPythonApplication rec {
...
 
   checkPhase = ''
+    runHook preCheck
     $out/bin/simp_le --test
+    runHook postCheck
   '';
 
-  propagatedBuildInputs = with python3Packages; [ acme setuptools-scm josepy idna ];
+
+  propagatedBuildInputs = with python3Packages; [ acme setuptools-scm josepy idna setuptools ];
```
Comment (mweinelt) [inline pkgs/tools/admin/simp_le/default.nix:27]: Please move setuptools into nativeBuildInputs, unless simp_le imports `pkg_resources`.

**Comment by mweinelt** [comment]
@ofborg build simp_le

File: pkgs/tools/admin/simp_le/default.nix (hunk 3 above)
Comment (mweinelt) [reply pkgs/tools/admin/simp_le/default.nix:27]: This is still unaddressed. I doubt that `setuptools` is a runtime dependency.

Comment (makefu) [reply pkgs/tools/admin/simp_le/default.nix:27]: well your doubt may be missguided:
setuptools in nativeinputs:
```
~/nixpkgs result/bin/simp_le 
Traceback (most recent call last):
  File "/nix/store/3lqjn67l5ncfp0a96rgc0qfgi21sjvb2-simp_le-client-0.20.0/bin/.simp_le-wrapped", line 6, in <module>
    from simp_le import main
  File "/nix/store/3lqjn67l5ncfp0a96rgc0qfgi21sjvb2-simp_le-client-0.20.0/lib/python3.11/site-packages/simp_le.py", line 41, in <module>
    import pkg_resources
ModuleNotFoundError: No module named 'pkg_resources'
```

**Comment by makefu** [comment]
added new runtime dependencies according to https://github.com/zenhack/simp_le/blob/master/setup.py#L9-L30

File: pkgs/tools/admin/simp_le/default.nix (hunk 3 above)
Comment (mweinelt) [reply pkgs/tools/admin/simp_le/default.nix]: Five kinds of weird, but okay!

File: pkgs/tools/admin/simp_le/default.nix [hunk 4]
```diff
@@ -17,12 +18,13 @@ python3Packages.buildPythonApplication rec {
...
   checkPhase = ''
+    runHook preCheck
     $out/bin/simp_le --test
+    runHook postCheck
   '';
 
-  propagatedBuildInputs = with python3Packages; [ acme setuptools-scm josepy idna ];
-
-  nativeCheckInputs = with python3Packages; [ mock ];
+  # both setuptools-scm with pkg_resources and mock are runtime dependencies
```
Comment (SuperSandro2000) [inline pkgs/tools/admin/simp_le/default.nix:26]: pkg_resources is part of just setuptools

**Pull request merged**